modules/git/repo_commit.go

```diff
@@ -161,7 +161,7 @@ func (repo *Repository) searchCommits(id SHA1, opts SearchCommitsOptions) ([]*Co
 				// add previous arguments except for --grep and --all
 				hashCmd.AddArguments(args...)
 				// add keyword as <commit>
-				hashCmd.AddArguments(v)
+				hashCmd.AddArguments("--end-of-options", v)
```

Comment:
This one is a bit more difficult. What is `v` supposed to be here? Presumably it is supposed to be a refish and not a path?

If we're happy for `v`s with initial `-` to be only recognised as refishs then checking if it has an initial `-` and then prefixing with `git.BranchPrefix` would be acceptable. If we want to allow paths with initial `-` too we'd need to run an additional git log with `git log ... -- v`.

Reply:
it represent's a commit, so we can parse if it's falid hex